### ISSUE TITLE
Android ServiceWorkerControllerCompat.setServiceWorkerClient(null) makes Webivew Plugin Crashes

### ISSUE THREAD

**Issue by ycv005** [pichillilorenzo/flutter_inappwebview#1151]
- [x] I have read the [Get Started - Installation](https://inappwebview.dev/docs/get-started/installation/) section
- [x] I have read and done the [Get Started - Setup Android](https://inappwebview.dev/docs/get-started/setup-android/) section
- [x] I have read and done the [Get Started - Setup iOS](https://inappwebview.dev/docs/get-started/setup-ios/) section
- [x] I have already searched for the same problem

## Environment

| Technology           | Version       |
| -------------------- | ------------- |
| Flutter version      |    2.10.5   |
| Plugin version       |     flutter_inappwebview: 5.4.1+1    |
| Android version      |    12    |
| iOS version          |          NA     |
| Xcode version        |      NA       |

Device information: 

Realme GT Master Edition

## Description

After update the Webview Plugin getting below error  (The Error Log is very long and is not readable. I have shared the readable part)

```
E/cr_AwBgThreadClient(10843): Client raised exception in shouldInterceptRequest. Re-throwing on UI thread.
E/cr_AwBgThreadClient(10843): The following exception was raised by shouldInterceptRequest:
D/AndroidRuntime(10843): Shutting down VM
E/AndroidRuntime(10843): FATAL EXCEPTION: main
E/AndroidRuntime(10843): Process: com.example.tbo, PID: 10843
E/AndroidRuntime(10843): java.lang.NullPointerException: Attempt to invoke virtual method 'android.webkit.WebResourceResponse androidx.webkit.ServiceWorkerClientCompat.shouldInterceptRequest(android.webkit.WebResourceRequest)' on a null object reference
E/AndroidRuntime(10843): 	at androidx.webkit.internal.FrameworkServiceWorkerClient.shouldInterceptRequest(FrameworkServiceWorkerClient.java:45)
E/AndroidRuntime(10843): 	at Sf0.a(chromium-TrichromeWebViewGoogle6432.aab-stable-489612734:32)
E/AndroidRuntime(10843): 	at Sa.a(chromium-TrichromeWebViewGoogle6432.aab-stable-489612734:122)
E/AndroidRuntime(10843): 	at org.chromium.android_webview.AwContentsBackgroundThreadClient.shouldInterceptRequestFromNative(chromium-TrichromeWebViewGoogle6432.aab-stable-489612734:31)
2
F/crashpad(15624): -----BEGIN CRASHPAD MINIDUMP-----
F/crashpad(15624): )iyJ"&cB3vRwF=78_8Q_"WjDe_7,&~]&iJ7:YRwz':=~;oc;',2V{Arm%9!ZBVSt=XCt8[j*$nkZNw|u!b4SJg;C|5*Sf6zTOJFJ0k$G{udlZr2KvlF.`I{E%LN7dP3Ri#Fv]I-#Vw@K?rst;2n:%k`^Q760O7q#zaU&6a"'n}(xrNpl$+%M3dkNx&lf4D",4R#\dd3b]QE8qc-_y#v[wFb,tm~j],}d#f()Jl7S}g{+zl'OBuw)Vy?O]N/gO`oHbp_n8iSfXU}_gsfMK)p|o{WN[PIxyPplYuA5qHjsH5lmqL6CF2N*jm'f/Zdr4^BnoC|7_2dL4h[fFiOfv1].QfV0yPoj$K1]gAtcPD!UT+BofTyu<OL%lmleSLbPaPvQ:h#2l;D1*6oPA#JMAe0.$A3i|PET~W5B$lmT6&danTj>9)o;ma0hg.t5gbMZJ)8nCov@ra4Hy[X@V[oZ5TSRRvZykNyue`|vNiYfx>2jP7qj4R"a/_ODqcODR0}G:_b+
F/crashpad(15624): CsXK3b.Wkknd2/?6rrjs;,aP?n&w@QX@XF=IZ2Z^Wt-YBU2h)6.N|J+e'F9]I&}n"FDBW/ML<|O8~WfCK_6_lm/\--DIRaBDHu.-NnrA96u.';<)gN|7E0:r=d_L,h(>zaK.soDdJ,9(~sPUO(fB9n[57Ty99gR]Q&&]an`b+A>EoA&1dFy=FJn;zqiupZ(N!NBJh:+f&qqLUf)Az53J6L|MuZ4q&D6]QPo;5;gsOGk=iu\IQ7vZju)ddwzA8`1*aY2`w-h!6O~SP\4JN?v/mN"O/7[:@hH.Es~C}h/g8a7/'H_+8<reVRLWiuyKHokm,AKY-w~*-SoS>shyy/^4!PEA)f\b7S;'#>MqTSS_)TzN;C}0kN}q$fZhlWxe6ZcP:Us|3J"JnW=s&1~Y?ccI><[sm2.eDC2,qOGfblw+2bOm[\#({l:6'psu&x~[#R(5eSq1Y=*+Uw>KH<@jPQGFTrNT&<ojEHfLg?^fju]nB>MdU<Hs%B,xTIxSh[uLa^sT
F/crashpad(15624): @9!7OI-A3\Xwl\u!?WMWNC[!YK?7ge>Q;V7W~tEJYd_;hvvcwRkAEL)*Grya1e&wLAuL`bxa~LSa%T:yEu-!&3owu>zsQnBLhL/cw,eZFscC3-lB;oBG_3%HX$].xo^<3_O5X}yu$UBJDgF/etZ(E?~Qq$K}r'\^EA#S{cN\UNES6aC0Es2}nW]hEKr]sQses]=fU6/(z2lIo6{OTBi(u\Pivb!8|OK>y,J<EKH2JGH*dsJA->Au./=/zHQvjz#4W4{VbgfvA\,M@.%4IUBopzG[EsO)TYh54lpI?@L[c2D(0l^F)ezn[2B&Doia9s@W]ZttR]APc.pohhRxgvPp2r_um>F)SF6P6Svp175<W)8lGA}p~duO]>d.<qr*}K}YoWtm@+("'K.ZwnT`'RNl#,s`GfxGx+6PZR4q6<EK7S}gBs&1&EO@^V;gLL2,vdh81u`TLK<7[w&V4qV,}KLopW@m,A~''lyYaoQL-R81\ME9in`r"oUBJYz&g@:AlgtV
```

**Expected behavior:** 

**Current behavior:** 

## Steps to reproduce

1. This
2. Than that
3. Then

**Comment by pichillilorenzo** [comment]
Are you using the Service Worker Controller API?

**Comment by ycv005** [comment]
@pichillilorenzo I am not aware of that, can you tell me where to check. For now, I can show you piece of code, executed my side.
I think introducing shareState might has cause the issue, beause before that, all was working fine.
```
 _browser.open(
        url: Uri.parse(redirectUrl),
        options: ChromeSafariBrowserClassOptions(
          android: AndroidChromeCustomTabsOptions(
            toolbarBackgroundColor: Color(int.parse(color)),
            // shareState: CustomTabsShareState.SHARE_STATE_OFF,
            addDefaultShareMenuItem: false,
            enableUrlBarHiding: true,
          ),
        ),
      );
```

**Comment by pichillilorenzo** [comment]
I mean if you are using the `AndroidServiceWorkerController` class somewhere inside of your code, such as my code example here: https://github.com/pichillilorenzo/flutter_inappwebview/blob/f511e4d1f5441f277231c9e05a70cc917755fb64/example/lib/main.dart#L31

**Comment by pichillilorenzo** [comment]
Ok, I found out what is the problem. I will fix it soon!

**Comment by pichillilorenzo** [comment]
Released new version `5.4.1+2` with the fix.
